serde-support/src/main/java/io/micronaut/serde/support/serdes/EnumSerde.java

```diff
@@ -67,6 +68,30 @@ public E deserializeNonNull(Decoder decoder, DecoderContext decoderContext, Argu
         }
     }
 
+    @Override
+    public Deserializer<E> createSpecific(DecoderContext context, Argument<? super E> type) {
+        try {
+            BeanIntrospection<? super E> si = introspections.getDeserializableIntrospection(type);
+            return (decoder, context1, type1) -> {
+                String s = decoder.decodeString();
```

Comment:
what if it's not String

Reply:
yes with valueOf only String made sense, but if we support JsonCreator, there should also be support for non-String inputs to that creator

Reply:
Ok implemented support for other types, but not sure how to implement support for multiple arguments so restricted to work argument